### ISSUE TITLE
`pkg_rpm` to `pkg_tar` contains duplicated rpm files

### ISSUE THREAD

**Issue by k0walik** [bazelbuild/rules_pkg#685]
I noticed inconsistent behaviour when we feed `pkg_deb` and `pkg_rpm` to `pkg_tar`:
- `pkg_deb` to `pkg_tar` contains only a single file that matches `package_file_name` attribute
- `pkg_rpm` to `pkg_tar` contains two files in tarball: one named after the rule, the other one that is named after `package_file_name`

While putting a single tarball may not have much sense, there are use-cases where one need to ship additional files along with deb/rpm package (i.e. instructions). My initial thought was the difference is lack of OutputGroupInfo in `pkg_rpm`, but I think the problem is `DefaultInfo.files` return value. `pkg_deb` returns only the output file, while `pkg_rpm` contains two output files:  the default output file and output file.

Minimum reproducible example:

```starlark
# //tests/deb_rpm_tar/BUILD
load("//pkg:deb.bzl", "pkg_deb")
load("//pkg:tar.bzl", "pkg_tar")
load("//pkg:rpm.bzl", "pkg_rpm")
load("//pkg:mappings.bzl", "pkg_files")

genrule(
    name = "gen_foo",
    outs = ["usr/bin/foo"],
    cmd = "touch $@",
)

pkg_tar(
    name = "deb_input",
    srcs = [":gen_foo"],
)

pkg_deb(
    name = "test_deb",
    data = ":deb_input",
    description = "Some package",
    maintainer = "john.doe@example.com",
    package = "debpackage",
    package_file_name = "debpackage-0.0.0.deb",
    version = "0.0.0",
)

pkg_tar(
    name = "deb_tar",
    srcs = [":test_deb"],
)

pkg_files(
    name = "rpm_input",
    srcs = [":gen_foo"],
)

pkg_rpm(
    name = "test_rpm",
    srcs = [":rpm_input"],
    description = "Description",
    license = "Apache 2.0",
    package_file_name = "rpmpackage-0.0.0-0.rpm",
    release = "1",
    summary = "Summary",
    version = "0.0.0",
)

pkg_tar(
    name = "rpm_tar",
    srcs = [":test_rpm"],
)
```

```bash
$ tar tvf bazel-out/k8-fastbuild/bin/tests/deb_rpm_tar/deb_tar.tar
-r-xr-xr-x 0/0           10662 2000-01-01 00:00 debpackage-0.0.0.deb
$ tar tvf bazel-out/k8-fastbuild/bin/tests/deb_rpm_tar/rpm_tar.tar 
-r-xr-xr-x 0/0            6254 2000-01-01 00:00 rpmpackage-0.0.0-1.rpm
-r-xr-xr-x 0/0            6254 2000-01-01 00:00 test_rpm.rpm
```

**Comment by aiuto** [comment]
I think the root cause is really`package_file_name = "rpmpackage-0.0.0-0.rpm"`.
That will give you an implicit output of a real file "rpmpackage-0.0.0-0.rpm"
and a default output of the symlink "test_rpm.rpm".   This is an intentional behavior:
- the physical file has a name containing the right (possibly dynamically determined) version numbers
- rules depending on the rule name can pick up the rpm at a well known name.

The same kind of thing happens for pkg_deb(). Anyone depending on just the name gets all output files.

If you need to get a specific single file, you can do the file_group+output_group redirection as seen here.
https://github.com/bazelbuild/rules_pkg/blob/edd4d3d643f01e6ac5cb529fc9d7e01dc8c9367e/examples/where_is_my_output/BUILD#L45

So in #684 you have the core of a correct change, except the tests don't test the right thing - the number of files is variable, not fixed. What it should be is that the expected output groups are created in particular situations.  That should be an analysis test or two. I can comment more about how do to that later. But now I ran out of time on rules_pkg for the week.

**Comment by k0walik** [comment]
I think this is still inconsistent behaviour between `pkg_deb` and `pkg_rpm` rules. If the output should contain both `package_file_name.rpm` and symlink to `rule_name.rpm`, then the `pkg_deb` should be fixed to include the symlink as well - not just the `package_file_name.deb`. See the return value implementation here: https://github.com/bazelbuild/rules_pkg/blob/main/pkg/private/deb/deb.bzl#L168
That being said: can I piggy back it on #684 or do we want another change to address this?

As for the testing part I'll update my change to test it correctly.

**Comment by aiuto** [comment]
I stand corrected. The DefaultInfo for the other rules is to have only the single file, not both.
Forget the tests, they should fold into the existing file naming tests and or be analysis tests.
Changing RPM to agree with the other rules should be a separate change.

**Comment by aiuto** [comment]
Is this fixed by #684

**Comment by k0walik** [comment]
Fixed in #692
